src/build_options.rs

```diff
@@ -23,6 +24,25 @@ use std::path::PathBuf;
 // and more restrictive.
 const PYO3_BINDING_CRATES: [&str; 2] = ["pyo3-ffi", "pyo3"];
 
+#[allow(clippy::type_complexity)]
+pub static PYO3_MINIMUM_PYTHON_MINOR: Lazy<HashMap<&'static str, HashMap<(u64, u64), usize>>> =
+    Lazy::new(|| {
+        let mut map = HashMap::new();
+
+        let mut pyo3 = HashMap::new();
+        pyo3.insert((0, 12), 6);
+        pyo3.insert((0, 13), 6);
+        pyo3.insert((0, 14), 6);
+        pyo3.insert((0, 15), 6);
+        pyo3.insert((0, 16), 7);
+        map.insert("pyo3", pyo3);
+
+        let mut pyo3_ffi = HashMap::new();
+        pyo3_ffi.insert((0, 16), 7);
+        map.insert("pyo3-ffi", pyo3_ffi);
+        map
+    });
```

Comment:
I might be tempted to implement this as a function which assumes upper bounds. That way this only needs updating when PyO3 changes minimum version rather than every release.

```suggestion
pub fn pyo3_minimum_python_version(major_version: u8, minor_version: u8) -> Option<(u8, u8)> {
    if (major_version, minor_version) >= (0, 16) {
        Some((3, 7))
    } else if (major_version, minor_version) >= (0, 12) {
        Some((3, 6))
    } else {
        None
    }
 } 
 
pub fn pyo3_ffi_minimum_python_version(major_version: u8, minor_version: u8) -> Option<(u8, u8)> {
    if (major_version, minor_version) >= (0, 16) {
        pyo3_minimum_python_version(major_version, minor_version)
    } else {
        None
    }
 }
```